client/ayon_core/hosts/maya/plugins/publish/validate_alembic_options_defaults.py

```diff
@@ -0,0 +1,91 @@
+import pyblish.api
+
+from ayon_core.pipeline import OptionalPyblishPluginMixin
+from ayon_core.pipeline.publish import RepairAction, PublishValidationError
+
+
+class ValidateAlembicOptionsDefaults(
+    pyblish.api.InstancePlugin, OptionalPyblishPluginMixin
+):
+    """Validate the attributes on the instance are defaults.
+
+    The defaults are defined in the project settings.
+    """
+
+    order = pyblish.api.ValidatorOrder
+    families = ["pointcache", "animation"]
+    hosts = ["maya"]
+    label = "Validate Alembic Options Defaults"
+    actions = [RepairAction]
+    optional = True
+
+    @classmethod
+    def _get_plugin_name(cls, publish_attributes):
+        for key in ["ExtractAnimation", "ExtractAlembic"]:
+            if key in publish_attributes:
+                return key
+
+    @classmethod
+    def _get_settings(cls, context):
+        maya_settings = context.data["project_settings"]["maya"]
+        settings = maya_settings["publish"]["ExtractAlembic"]
+        return settings
+
+    @classmethod
+    def _get_publish_attributes(cls, instance):
```

Comment:
this looks like a duplicate of `get_attr_values_from_data_for_plugin`.

Reply:
It does indeed - but it does so by string instead of plugin `class` which can't really be used here since this Validator in particular wouldn't have access to the `ExtractAlembic` and `ExtractAnimation` classes because usually these plug-ins aren't on pythonpath as an importable file, right?

Anyway, plus the fact that this decides on which of the two plug-in it should apply to instead of running for one of them.

Admittedly, it's definitely not a nice pattern what is done here that way. The only other way I see if creating two Validator plug-ins that dedicate to run over `animation` and `pointcache` family individually. Then at least it's clearer which family belong to which extractors and we do not need the "dynamic" decision on which of the two to use to get the settings from?

Reply:
This validator has been split now.